biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -41,15 +42,23 @@
 	}
 
 	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port) throws Exception {
-		while (true)
+		connect(agent, supervisor, host, port, null);
+	}
+
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, Integer timeout)
```

Comment:
Using Integer, to get null as a possible input, means a lot of boxing/unboxing for comparisons and `-=` below. Can't we use -1 to mean no timeout?